cvrf2csaf/cvrf2csaf.py

```diff
@@ -74,16 +74,24 @@ def _compose_final_csaf(self) -> dict:
         # Merges first level leaves into final CSAF document.
         # [mapping table](https://github.com/tschmidtb51/csaf/blob/csaf-2.0-what-is-new-table/notes/whats-new-csaf-v2.0-cn01.md#e4-mapped-elements)
 
+        # Mandatory root elements
         final_csaf = {'document': {}}
         final_csaf['document'] = self.document_leaf_elements.csaf
-        if self.document_acknowledgments.csaf:
-            final_csaf['document']['acknowledgments'] = self.document_acknowledgments.csaf
-        final_csaf['document']['notes'] = self.document_notes.csaf
         final_csaf['document']['publisher'] = self.document_publisher.csaf
         final_csaf['document']['tracking'] = self.document_tracking.csaf
-        final_csaf['document']['references'] = self.document_references.csaf
-        final_csaf['product_tree'] = self.product_tree.csaf
-        final_csaf['vulnerabilities'] = self.vulnerability.csaf
+
+        # Optional root elements
+        if self.document_notes.csaf:
```

Comment:
A more data centered approach might do this:
```python
        mappings = (
            (final_csaf['document'], 'notes, self.document_notes.csaf),
            (final_csaf['document'], 'references', self.document_references.csaf),
            (final_csaf['document'], 'acknowledgments', self.document_acknowledgments.csaf),
            (final_csaf, 'product_tree', self.product_tree.csaf),
            (final_csaf, 'vulnerabilities', self.vulnerability.csaf),
        )
        for target, slot, value in mappings:
            if value:
                target[slot] = value
```

Short example test:
```python
>>> t = {'a': {}}
>>> b = None  # no insertion of b value
>>> c = True
>>> d = 42
>>> e = False  #no insertion of e value
>>> mappings = ((t['a'], 'bkey', b), (t['a'], 'ckey', c), (t, 'dkey', d), (t, 'ekey', e))
>>> for target, slot, value in mappings:
...     if value:
...         target[slot] = value
...
>>> t
{'a': {'ckey': True}, 'dkey': 42}
```

... just saying - all well as it is already.

Reply:
Nice one :) implemented